oauth2_http/java/com/google/auth/oauth2/ExternalAccountSupplierContext.java

```diff
@@ -0,0 +1,78 @@
+package com.google.auth.oauth2;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import java.io.Serializable;
+
+/** Context object to pass relevant variables from external account credentials to suppliers. */
+public class ExternalAccountSupplierContext implements Serializable {
+
+  private static final long serialVersionUID = -7852130853542313494L;
+
+  private final String audience;
+  private final String subjectTokenType;
+
+  /** Internal constructor. See {@link ExternalAccountSupplierContext.Builder}. */
+  ExternalAccountSupplierContext(Builder builder) {
+    this.audience = builder.audience;
+    this.subjectTokenType = builder.subjectTokenType;
+  }
+
+  /**
+   * Gets the credentials expected audience.
+   *
+   * @return the audience.
+   */
+  public String getAudience() {
+    return audience;
+  }
+
+  /**
+   * Gets the credentials expected subject token type.
+   *
+   * @return the subject token type.
+   */
+  public String getSubjectTokenType() {
+    return subjectTokenType;
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  /** Builder for external account supplier context. */
+  public static class Builder {
```

Comment:
Should this be private? Credentials handle initializing the context and passing it, when would a user have to?

Reply:
A user generally wouldn't have to, but it seems weird to have a public class that there is no way to construct publicly

Reply:
Actually a user would want to be able to build these for unit testing of their supplier functions

Reply:
I think it's fine to have a public class that they can't construct because the library handles creating it and passing it to them. For their unit tests they can just mock it.